dataflow/src/main/java/org/checkerframework/dataflow/cfg/node/ObjectCreationNode.java

```diff
@@ -88,19 +163,37 @@ public boolean equals(@Nullable Object obj) {
         if (constructor == null && other.getConstructor() != null) {
             return false;
         }
+        if ((receiver != null && other.getReceiver() == null)
+                || (receiver == null && other.getReceiver() != null)
```

Comment:
I think you can make this similar to how `constructor` is handled above: here only check `receiver == null && other.getReceiver() != null` and then below in the return add `getReceiver().equals(other.getReceiver())`.